### DISCUSSION TITLE
Docker build failing due to schemas being out of date

### DISCUSSION THREAD

**Discussion by EwanLyon** [keystonejs/keystone#7346] [Questions] [answered]
### Reproduction
Clone https://github.com/ausspeedruns/ausspeedruns.com/tree/65e4a21c07fddb6542d7a2f9057edb75c8a8fd52. Specifically the `keystonejs` branch **and now specifically the `65e4a21c07` commit**.
Run `docker build -t ausruns/website . --build-arg SESSION_SECRET="AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"`
*The command is missing a build arg for the database url but we hit the error prior to where it would be needed.*

The docker container will fail when building the site as both schema's are out of date.
The command run prior to the build is `keystone postinstall --fix`.

### Expectation
The docker container recognises the schemas just built or the build runs --fix regardless.

### Screenshot
Context: `npm run update-keystone` runs `keystone postinstall --fix`
![image](https://user-images.githubusercontent.com/19513845/157862246-2d7eb0c5-01c9-4689-b5f6-8cbdffebb904.png)

**Comment by EwanLyon** [comment]
Ok I figured out my exact problem was that I was assuming that docker would run with `NODE_ENV="production"` when building. Adding that solved the issue as I have a check to see whether to use azure files or local files.

However the issue I find still stands where running `keystone postinstall --fix` wasn't being seen by the build step or that the build step should run `--fix` regardless.

**Comment by dcousens** [comment, marked as answer]
Hi @EwanLyon,

It's recommended that you only use 1 type of database for development and production.

As fixed by you in https://github.com/ausspeedruns/ausspeedruns.com/commit/48778da7869d74a3258c5a3a42e137a5a8faa55f,  this problem came about as a a result of you having different database providers for `NODE_ENV="production"` and development.
